File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int counts[26] = {0};
        int l = 0, maxLen = 0, maxFreq = 0;
        for(int r = 0; r < s.length(); r++)
        {
            counts[s[r] - 'A']++;
            maxFreq = max(maxFreq, counts[s[r] - 'A']);

            if((r - l + 1) - maxFreq > k)
            {
                counts[s[l] - 'A']--;
                l++;
            }
            
            maxLen = max(maxLen, r - l + 1);
        }

        return maxLen;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        l = 0
        r = 0

        while (l and r) < len(prices):
            add = prices[r] - prices[l]
            profit = max(profit, add)
            if prices[l] > prices[r]:
                l = r
                r += 1
            r += 1
        return profit
